library/lib-core-logging/src/main/java/com/sokima/lib/corelogging/PatternMaskingLayout.java

```diff
@@ -0,0 +1,41 @@
+package com.sokima.lib.corelogging;
+
+import ch.qos.logback.classic.PatternLayout;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import com.sokima.lib.corelogging.usecase.MaskUseCase;
+import com.sokima.lib.corelogging.usecase.impl.JsonMaskUseCaseImpl;
+import com.sokima.lib.corelogging.usecase.impl.ToStringMaskUseCaseImpl;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PatternMaskingLayout extends PatternLayout {
```

Comment:
use final as much as possible. 

A few explanations: 
1 - all local vars, method params, constructor params should be final to prevent Side Affects (refer to functional programming what it is like Scala, Haskell, Rust, ..). Why in lambda you can use only `efficiently final` vars?

2 - all classes should be defined as final, unless they are designed for inheritance. Be aware with Framework classes, e.g. Entity classes in Hibernate shouldn't be final because of issues. 

3 - mark as final all protected/class methods, which are not going to be overridden

4 - Marking with `final`  tells future editors of the code something about your intentions.

5 - Making all immutable makes asynchronous execution more predictable

And much more, feel free to explore topic on your own.
P. S. Java should have made final the default, but this is issue in java architecture. Moder Programming Languages have this feature as default.